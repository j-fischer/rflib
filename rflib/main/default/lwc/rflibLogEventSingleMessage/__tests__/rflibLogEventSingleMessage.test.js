import { createElement } from 'lwc';
import RflibLogEventSingleMessage from 'c/rflibLogEventSingleMessage';

describe('c-rflib-log-event-single-message', () => {
    afterEach(() => {
        // The jsdom instance is shared across test cases in a single file so reset the DOM
        while (document.body.firstChild) {
            document.body.removeChild(document.body.firstChild);
        }
    });

    async function setupElementComponent(props = {}) {
        const element = createElement('c-rflib-log-event-single-message', {
            is: RflibLogEventSingleMessage
        });
        // Ensure fieldVisibility is properly initialized
        console.log(props);
        if (props.fieldVisibility) {
            console.log(props.fieldVisibility);
            element.fieldVisibility = { ...props.fieldVisibility };
        }
        if (props.message) {
            console.log(props.message);
            element.message = { ...props.message };
        }
        document.body.appendChild(element);
        await Promise.resolve(); // Wait for the component to render
        return element;
    }

    it('Verify that the component handles JSON message rendering correctly.', () => {
        // Arrange
        const element = createElement('c-rflib-log-event-single-message', {
            is: RflibLogEventSingleMessage
        });
        element.message = {
            content: '{"key":"value"}',
            preview: 'JSON Preview',
            isJson: true,
            isText: false,
            formattedJson: '{"key":"value"}'
        };

        // Act
        document.body.appendChild(element);

        // Assert
        const jsonLabel = element.shadowRoot.querySelector('.json-label');
        expect(jsonLabel.textContent).toBe('JSON Preview');
    });

    it('Verify that the component handles text message with date.', async () => {
        // Arrange
        const message = {
            content: '2025-05-23T20:01:46.515Z|DEBUG|LogEventMonitor|Registering Error Listener',
            isJson: false,
            isText: true
        };
        const fieldVisibility = {
            showDate: true,
            showLogLevel: false,
            showContext: false,
            showCreatedBy: false
        };
        let element = await setupElementComponent({message: message, fieldVisibility: fieldVisibility});

        document.body.appendChild(element);
        await new Promise(resolve => setTimeout(resolve, 0));
        const dateText = element.shadowRoot.querySelectorAll('.text-message');
        const formattedText = dateText[0].querySelector('lightning-formatted-text');
        expect(formattedText.value).toBe('2025-05-23T20:01:46.515Z|Registering Error Listener');
    });

    it('Verify that the component handles text message log level rendering correctly.', async () => {
        // Arrange
        const message = {
            content: '2025-05-23T20:01:46.515Z|DEBUG|LogEventMonitor|Registering Error Listener',
            isJson: false,
            isText: true
        };
        const fieldVisibility = {
            showDate: false,
            showLogLevel: false,
            showContext: true,
            showCreatedBy: false
        };
        let element = await setupElementComponent({message: message, fieldVisibility: fieldVisibility});

        document.body.appendChild(element);
        await new Promise(resolve => setTimeout(resolve, 0));
        const dateText = element.shadowRoot.querySelectorAll('.text-message');
        const formattedText = dateText[0].querySelector('lightning-formatted-text');
        expect(formattedText.value).toBe('LogEventMonitor|Registering Error Listener');
    });

    it('Verify that the component expands and collapses JSON content correctly.', async () => {
        // Arrange
        const message = {
            content: '{"key":"value"}',
            preview: 'JSON Preview',
            isJson: true,
            isText: false,
            formattedJson: '{"key":"value"}'
        };
        const fieldVisibility = {
            showDate: true, 
            showLogLevel: true, 
            showContext: true, 
            showCreatedBy: true
        };
        const element = await setupElementComponent({message: message, fieldVisibility: fieldVisibility});


        // Act
        document.body.appendChild(element);

        // Assert initial state (collapsed)
        let jsonContent = element.shadowRoot.querySelector('.json-content');
        expect(jsonContent).toBeNull();
        await new Promise(resolve => setTimeout(resolve, 0));

        // Act: Click to expand
        const button = element.shadowRoot.querySelector('lightning-button-icon');
        button.click();
        await new Promise(resolve => setTimeout(resolve, 0));

        // Assert expanded state
        jsonContent = element.shadowRoot.querySelector('.json-content');
        expect(jsonContent).not.toBeNull();

        // Act: Click to collapse
        button.click();
        await new Promise(resolve => setTimeout(resolve, 0));

        // Assert collapsed state
        jsonContent = element.shadowRoot.querySelector('.json-content');
        expect(jsonContent).toBeNull();
    });
});