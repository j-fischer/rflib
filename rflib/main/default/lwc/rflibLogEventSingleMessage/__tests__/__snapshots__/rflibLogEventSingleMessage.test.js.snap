// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`c-rflib-log-event-single-message Verify that the component renders correctly with default properties. 1`] = `
<c-rflib-log-event-single-message>
  #shadow-root(open)
    <style
      type="text/css"
    >
      .json-message {background-color: #f8f9fa;border: 1px solid #dee2e6;border-radius: 0.375rem;overflow: hidden;}.json-header {display: flex;align-items: center;padding: 0.5rem 0.75rem;background-color: #e9ecef;border-bottom: 1px solid #dee2e6;cursor: pointer;transition: background-color 0.2s ease;}.json-header:hover {background-color: #dde0e3;}.json-label {font-weight: 600;font-size: 0.875rem;color: #495057;margin-left: 0.5rem;white-space: nowrap;overflow: hidden;text-overflow: ellipsis;display: inline-block;width: 100%;}.json-content {margin: 0;padding: 0.75rem;background-color: #f8f9fa;font-family: 'Monaco', 'Menlo', 'Ubuntu Mono', 'Consolas', 'source-code-pro', monospace;font-size: 0.8125rem;line-height: 1.4;color: #212529;white-space: pre-wrap;word-wrap: break-word;border: none;border-radius: 0;overflow-x: auto;}.json-content code {background: none;color: inherit;font-size: inherit;padding: 0;}.json-preview {padding: 0.75rem;font-style: italic;color: #6c757d;font-size: 0.875rem;background-color: #ffffff;}.text-message {padding: 0.25rem 0;line-height: 1.5;}.text-message lightning-formatted-text {white-space: pre-wrap;word-wrap: break-word;}
    </style>
    <div
      class="log-message-item"
    >
      <div
        class="text-message"
      >
        <lightning-formatted-text>
          #shadow-root(open)
        </lightning-formatted-text>
        <lightning-formatted-text>
          #shadow-root(open)
        </lightning-formatted-text>
        <lightning-formatted-text>
          #shadow-root(open)
        </lightning-formatted-text>
        <lightning-formatted-text>
          #shadow-root(open)
        </lightning-formatted-text>
        <lightning-formatted-text>
          #shadow-root(open)
        </lightning-formatted-text>
      </div>
    </div>
</c-rflib-log-event-single-message>
`;
